adaptive_memory_v4.0.py

```
@@ -0,0 +1,2251 @@
+                    # Format: [Tags: tag1, tag2] Content [Memory Bank: Bank] [Confidence: X.XX]
+                    tags_str = ", ".join(tags) if tags else "none"
+                    confidence = op.get("confidence", 1.0)
+
+                    final_content = f"[Tags: {tags_str}] {content} [Memory Bank: {bank}] [Confidence: {confidence:.2f}]"
+
+                    # Add memory - using Router add_memory for Vector Indexing
+                    try:
+                        mem_obj = None
+                        try:
+                            # Try the high-level router function which handles vector indexing
+                            logger.info(f"Attempting to add memory via router add_memory (Vector-Aware)...")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)